synapse/models/syn.py

```diff
@@ -202,3 +204,26 @@ def _revModl201710191144(self):
                     i = i + len(chunk)
                     logger.debug('Loading {:,d} [{}%] rows into transaction'.format(i, int((i / tot) * 100)))
         logger.debug('Finished adding node:created rows to the Cortex')
+
+    @modelrev('syn', 201711012123)
+    def _revModl201711012123(self):
+        now = s_common.now()
+        for form in self.core.getTufoForms():
+            adds = []
+            logger.debug('Computing all node:ndef values for{}'.format(form))
+            for i, p, v, t in self.core.store.getRowsByProp(form):
+                # This is quicker than going through the norm process
+                nv = s_common.guid((p, v))
+                adds.append((i, 'node:ndef', nv, now))
+
+            if adds:
+                tot = len(adds)
+                logger.debug('Adding {:,d} node:ndef rows for {}'.format(tot, form))
+                i = 0
+                n = 100000
+                for chunk in s_common.chunks(adds, n):
```

Comment:
since you've got these all in ram, we might as well let the storage layer try to do them all at once.  ( in one big addRows() call )